File: backend/src/main/java/net/tape/model/Subcategory.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class Subcategory implements Serializable {
    private Long id;
    private Long categoryId;
    private String categoryName;
    private String categorySlug;
    private String slug;
    private String name;
    private String blurb;
    private String imageUrl;
    private int sort;
    private int videoCount;
    private java.util.List<Video> videos;
}

File: web/src/api/components/schemas/Subcategory.ts
import { Video, VideoIO } from '../../components/schemas/Video';
import { Option } from 'fp-ts/lib/Option';
import { Int, string, array, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type Subcategory = {
	id: Option<Int>;
	categoryId: Option<Int>;
	categoryName: Option<string>;
	categorySlug: Option<string>;
	slug: Option<string>;
	name: Option<string>;
	blurb: Option<string>;
	imageUrl: Option<string>;
	sort: Option<Int>;
	videoCount: Option<Int>;
	videos: Option<Array<Video>>;
};
export const SubcategoryIO = type(
	{
		id: optionFromNullable(Int),
		categoryId: optionFromNullable(Int),
		categoryName: optionFromNullable(string),
		categorySlug: optionFromNullable(string),
		slug: optionFromNullable(string),
		name: optionFromNullable(string),
		blurb: optionFromNullable(string),
		imageUrl: optionFromNullable(string),
		sort: optionFromNullable(Int),
		videoCount: optionFromNullable(Int),
		videos: optionFromNullable(array(VideoIO)),
	},
	'Subcategory',
);
